### PR TITLE
Added script build/publish docker image

### PR DESCRIPTION
**Pull request by mikaelhm**
This PR is part of a larger change were users will use published docker images instead of building the image locally.

As this PR changed the Dockerfile, I fixed the local devcontainer environment as well.

### CODE REVIEW

File: scripts/build-docker-image.sh [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+# Don't run `docker build` and `docker image prune` when `--skip-build` in arguments
+[[ ${*/--skip-build//} != "${*}" ]] || {
+    docker build "${BUILD_ARGS[@]}" --build-arg TARGETPLATFORM="$TARGET_PLATFORM_TYPE"  --build-arg GIT_SHA="$GIT_REV" --build-arg VERSION="$VERSION" -t "$GHCR_ORG/$ORG/$IMAGE:$VERSION" .
+    docker image prune --force
+}
+
+[[ ${*/--latest//} != "${*}" ]] && {
+    docker tag "$GHCR_ORG"/"$ORG"/"$IMAGE":"$VERSION" "$GHCR_ORG"/"$ORG"/"$IMAGE":latest
+}
```
Comment (ccruzagralopes) [inline scripts/build-docker-image.sh:91]: I noticed that you used `"$GHCR_ORG/$ORG/$IMAGE:$VERSION"` here and then `"$GHCR_ORG"/"$ORG"/"$IMAGE":"$VERSION"` (quotes between each slash) in the following lines. Is there a difference?

File: scripts/build-docker-image.sh [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+[[ ${*/--squash//} != "${*}" ]] && {
+    command -v docker-squash >/dev/null &&
+        docker-squash "$GHCR_ORG"/"$ORG"/"$IMAGE":"$VERSION" -t "$GHCR_ORG"/"$ORG"/"$IMAGE":latest
```
Comment (ccruzagralopes) [inline scripts/build-docker-image.sh:95]: Should this use `latest` even without the --latest option? I noticed that you used a condition for other things

File: scripts/build-docker-image.sh [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        docker push "$GHCR_ORG"/"$ORG"/"$IMAGE":latest
+    }
+}
+
+[[ ${*/--clear//} != "${*}" ]] && {
+    docker rmi -f "$GHCR_ORG"/"$ORG"/"$IMAGE":"$VERSION"
+    [[ ${*/--latest//} != "${*}" ]] && {
+        docker rmi -f "$GHCR_ORG"/"$ORG"/"$IMAGE":latest
+    }
+}
```
Comment (ccruzagralopes) [inline scripts/build-docker-image.sh:113]: is the `--clear` option purposefully not in help?

File: scripts/build-docker-image.sh (hunk 1 above)
Comment (mikaelhm) [reply scripts/build-docker-image.sh:91]: I don't know actually this is based on the build script from the SDK repo:

https://github.com/project-chip/connectedhomeip/blob/master/integrations/docker/build.sh

File: scripts/build-docker-image.sh (hunk 3 above)
Comment (mikaelhm) [reply scripts/build-docker-image.sh:113]: No, I will add it.

File: scripts/build-docker-image.sh (hunk 2 above)
Comment (mikaelhm) [reply scripts/build-docker-image.sh:95]: Good point I will review it.

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -3,10 +3,8 @@
 // If you want to run as a non-root user in the container, see .devcontainer/docker-compose.yml.
 {
 	"name": "Backend: CHIP Test Harness",
-	"initializeCommand": "bash .devcontainer/initializeCommands.sh", // required to get the environment variables set for the sub-folder backend
+	// "initializeCommand": "bash .devcontainer/initializeCommands.sh", // required to get the environment variables set for the sub-folder backend
```
Comment (rquidute) [inline .devcontainer/devcontainer.json:6]: Since this line is commented, it should be removed?

File: scripts/build-docker-image.sh (hunk 1 above)
Comment (ccruzagralopes) [reply scripts/build-docker-image.sh]: oh, I see

Comment (mikaelhm) [reply scripts/build-docker-image.sh]: I standardized it :)

**Review by ccruzagralopes — APPROVED** [review]

**Review by hiltonlima — APPROVED** [review]

**Review by rquidute — APPROVED** [review]

**Pull request merged**